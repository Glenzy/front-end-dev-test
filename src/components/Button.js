import React from 'react';
import { buttonPropTypes } from '../types/productPropTypes';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

const button = (props) => {
    const {text, handleClick, type, classes, icon, name} = {...props}; 
    return (
        <button 
            type={type}
            className={`${classes}`} 
            onClick={handleClick}
            name={name}> 
                {icon && <FontAwesomeIcon icon={icon}  aria-label={type} />}
                {text} 
        </button>  
    );
}

button.propTypes = {
    ...buttonPropTypes,
}

export default button;