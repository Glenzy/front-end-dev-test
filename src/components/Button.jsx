import React from 'react';
import { buttonPropTypes } from '../types/productPropTypes';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

const button = (props) => {
    const {text, handleClick, type, classes, icon, name, iconPosition} = {...props}; 
    return (
        <button 
            type={type}
            className={`${classes}`} 
            onClick={handleClick}
            name={name}> 
                {icon && iconPosition == "left" && <FontAwesomeIcon icon={icon}  aria-label={type} />}
                {text}
                {type === "nav-link" && <span className="sr-only">(current)</span>} 
                 {icon && iconPosition == "right" && <FontAwesomeIcon icon={icon}  aria-label={type} />}
        </button>  
    );
}

button.propTypes = {
    ...buttonPropTypes,
}

export default button;