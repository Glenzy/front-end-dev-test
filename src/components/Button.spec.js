import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button.jsx';

let wrapper;

const handleClick = jest.fn();
const defaultProps = {
    handleClick,
  text: 'Test Button',
  type: 'primary',
  classes: 'test test-btn', 
  name: 'Test Button',
  icon:'cart-plus',
};

describe('the button component', ()=>{

  it('should call click handler when clicked', ()=>{
    givenButtonWithProps(defaultProps);
    wrapper.simulate('click');
    expect(handleClick.mock.calls.length).toBe(1);
  });

  it('renders correctly', () => {
    givenButtonWithProps(defaultProps);
    expect(wrapper).toMatchSnapshot();
  });
});

const givenButtonWithProps = (defaultProps) => wrapper = shallow(
  <Button {...defaultProps} />
);
