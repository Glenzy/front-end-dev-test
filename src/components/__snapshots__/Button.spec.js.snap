// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the button component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <button
    classes="test test-btn"
    handleClick={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    icon="cart-plus"
    name="Test Button"
    text="Test Button"
    type="primary"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <FontAwesomeIcon
          aria-label="primary"
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="cart-plus"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />,
        "Test Button",
        false,
      ],
      "className": "test test-btn",
      "name": "Test Button",
      "onClick": [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      },
      "type": "primary",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "aria-label": "primary",
          "border": false,
          "className": "",
          "fixedWidth": false,
          "flip": null,
          "icon": "cart-plus",
          "inverse": false,
          "listItem": false,
          "mask": null,
          "pull": null,
          "pulse": false,
          "rotation": null,
          "size": null,
          "spin": false,
          "symbol": false,
          "title": "",
          "transform": null,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "Test Button",
      false,
    ],
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <FontAwesomeIcon
            aria-label="primary"
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon="cart-plus"
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />,
          "Test Button",
          false,
        ],
        "className": "test test-btn",
        "name": "Test Button",
        "onClick": [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        },
        "type": "primary",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "aria-label": "primary",
            "border": false,
            "className": "",
            "fixedWidth": false,
            "flip": null,
            "icon": "cart-plus",
            "inverse": false,
            "listItem": false,
            "mask": null,
            "pull": null,
            "pulse": false,
            "rotation": null,
            "size": null,
            "spin": false,
            "symbol": false,
            "title": "",
            "transform": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "Test Button",
        false,
      ],
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
